com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/HDSubTarget.cs

```diff
@@ -85,28 +91,27 @@ internal virtual void MigrateTo(ShaderGraphVersion version)
         {
         }
 
+        /// <summary>
+        /// Override this method to handle subtarget specific migration (ie including private versioning)
+        /// in inherited subtargets
+        /// </summary>
+        internal virtual void Migrate()
+        {
+            if (migrationSteps.Migrate(this))
+                OnBeforeSerialize();
```

Comment:
Kept old behavior of calling OnBeForeSerialize() if migration happened but moved to own virtual function: this is to allow the most derived subtarget (plugin) to ensure all common subtarget versioning (HDRP's ShaderGraphVersion) migration is done before executing its own (private and unknown to HDRP) versioning line migration.